test/test_torch.py

```diff
@@ -12665,6 +12665,25 @@ def test_fill_diagonal(self):
         e1.fill_diagonal_(v, wrap=True)
         self.assertEqual(e1, e2)
 
+    def test_binary_op_input_output_overlap(self):
+        sz = 3
+        data = torch.randn(3 * sz)
+        for i in range(len(data) - sz):
```

Comment:
This is going to launch 6 * 6 * 6 = 216 multiplications, but we don't need to test all of these exhaustively because a lot of them are equivalent. It would be nice to just write down a few specific cases and then test those.